test/e2e/testsuites/testsuites.go

```diff
@@ -806,10 +806,52 @@ func (t *TestPod) SetupRawBlockVolume(pvc *v1.PersistentVolumeClaim, name, devic
 	t.pod.Spec.Volumes = append(t.pod.Spec.Volumes, volume)
 }
 
+func (t *TestPod) GetVolumeDiskURI(pvcname string) (string, error) {
+	pvc, err := t.client.CoreV1().PersistentVolumeClaims(t.namespace.Name).Get(context.TODO(), pvcname, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("Error getting pvc: %v", err)
+	}
+	pvname := pvc.Spec.VolumeName
+	pv, err := t.client.CoreV1().PersistentVolumes().Get(context.TODO(), pvname, metav1.GetOptions{})
+	if err != nil {
+		return "", fmt.Errorf("Error getting pv: %v", err)
+	}
+	var diskURI string
+	if pv.Spec.PersistentVolumeSource.CSI != nil {
+		diskURI = pv.Spec.PersistentVolumeSource.CSI.VolumeHandle
+	} else if pv.Spec.PersistentVolumeSource.AzureDisk != nil {
+		diskURI = pv.Spec.PersistentVolumeSource.AzureDisk.DataDiskURI
+	}
+
+	return diskURI, nil
+}
+
 func (t *TestPod) SetNodeSelector(nodeSelector map[string]string) {
 	t.pod.Spec.NodeSelector = nodeSelector
 }
 
+func (t *TestPod) ListNodes() []string {
+	var err error
+	var nodes *v1.NodeList
+	var nodeNames []string
+	nodes, err = t.client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	framework.ExpectNoError(err)
+	for _, item := range nodes.Items {
+		nodeNames = append(nodeNames, item.ObjectMeta.Name)
+	}
+	return nodeNames
+}
+
+func (t *TestPod) MakeNodeUnschedulable(nodeName string, makeUnschedulable bool) {
```

Comment:
Consider `SetNodeUnschedulable` since it is setting the state of the property and not unconditionally making the node unschedulable.